arch/posix/core/fatal.c

```diff
@@ -8,9 +8,12 @@
 #include <kernel.h>
 #include <arch/cpu.h>
 #include <kernel_structs.h>
-#include <misc/printk.h>
```

Comment:
Question: Is the intention that only the new logger subsystem be used at all?
So not only the old logger but also printk would be deprecated?

Reply:
Imo, they overlaps since printk is actually error message in kernel. Some may argue that it is not error but info. Then why not to use logger directly. In fact there is an option to redirect printk to logger but this is rather for backward compatibility. Direct use of the console in printk is also a drawback. As long as it is simple polling there is no problem, but once console over uart uses interrupts then it cannot be used in fault handler. Logger handles that (panic mode).

Reply:
Would it be reasonable to start by only enabling the logger panic in this PR, and let the user chose if he wants to use the old `printk` or route `printk` to the new logger with `CONFIG_LOG_PRINTK`?

Reply:
Maybe, but on the other hand replacing printk with LOG_.. adds value:
- LOG_... can be redirected to multiple backends so adding flashlog backend will immediately add crashlog. Printk can be redirected to log but then it takes more resources and is less efficient. 
- standard severity level configuration can be used instead of custom one (CONFIG_FAULT_DUMP)
- code is cleaner because #ifdef CONFIG_PRINTK can be removed.

Reply:
I'm not saying we should not do it eventually. My thinking was that this PR has 2 things:
* Activate the logger panic -> It makes sense to take this change in right away.
* Replacing calls from printk with logger. -> It's not urgent. Actually thanks to the CONFIG_LOG_PRINTK option, it could be done after the whole `SYS_LOG` => `LOG` transition. But starting to do the `printk` replacements today means losing them where the new logger does not yet work. So, if it were me, I would focus first on maturing the logger, getting in more backends for it, and doing the SYS_LOG=>LOG transition (that's already a lot of things to be done).

Reply:
Followed your suggestion and created #10098